models/sessions.js

```diff
@@ -0,0 +1,124 @@
+var Db        = require('../config/database');
+var mongoose  = require('mongoose');
+var crypto    = require('crypto')
+
+// Generates a random string to use as a session token
+function generateToken() {
+	return crypto.randomBytes(256).toString('base64');
+}
+
+// Create the schema for a session
+var SessionSchema = mongoose.Schema({
+
+	// Defaults to a string of 256 characters
+    token	  : { type : String, default : generateToken },
+
+    // Defaults to the current time and expires in 7 days
+    createdAt : { type : Date, default : Date.now, expires : '7d' },
+
+    // Holds the ObjectId of the user logged into this session
+    userId    : { type : mongoose.Schema.Types.ObjectId, required : true}
+});
+
+// Creates a collection named sessions in MongoDB
+var SessionMongoModel = Db.model('sessions', SessionSchema);
+
+// Takes a userId as an argument and returns the token string to the callback
+function findSessionByUserId(userId, callback) {
+	SessionMongoModel.findOne({ userId : userId }, function(err, session) {
+		if (err) {
+			// TODO: Error message?
+			callback(err, null);
+			return;
+		}
+		if (session)
+			callback(null, session.token);
+		else
+			callback({
+				code	: 400,
+				message : 'The session does not exist.'
+			}, null);
+	});
+}
+
+// Takes a token string as an argument and returns the userId to the callback
+function findUserByToken(clientToken, callback) {
+	SessionMongoModel.findOne({ token : clientToken }, function(err, session) {
+		if (err) {
+			// TODO: Error message?
+			callback(err, null);
+			return;
+		}
+		if (session)
+			callback(null, session.userId);
+		else
+			callback({
+				code	: 400,
+				message : 'The session does not exists.'
+			}, null);
+	});
+}
+
+// Creates a new session and attach user ObjectId to it
+function createSession(userId, callback) {
+	var newSession = new SessionMongoModel({ userId : userId });
+
+	newSession.save(function (err, session) {
+		if (err) {
+			// TODO: Error message?
+			callback(err, null);
+			return;
+		}
+		if (session)
+			callback(null, session);
```

Comment:
In the google docs we need to return both the token string and the TTL, which both belong to the session. We can do a callback with 3 arguments? I still need to figure out the TTL for the response.